### DISCUSSION TITLE
Realtime connection issues with wildcard cloudflare tunnel setup

### DISCUSSION THREAD

**Discussion by gmrizzo** [coollabsio/coolify#2825] [Q&A] [answered]
Hey guys, maybe someone can help me with this.
I have a wildcard domain setup with cloudflare tunnel. Everything works fine as configured except the realtime connection and I don't know why. 

Here are my configurations, I compared them to the documentation multiple times:

**Cloudflare Tunnel Domain Configuration:**
*.mydomain.com => http://localhost:80
mydomain.com => http://localhost:8000

**Cloudflare SSL/TLS Encryption Mode:** Full

**Coolify Config:**
.env:
    - PUSHER_HOST=realtime.mydomain.com
    - PUSHER_PORT=443

I restarted coolify with the script, the whole server and in the ui but connecting to the websocket doesn't work. The developer console states:
"WebSocket connection to 'wss://realtime.mydomain.com/app/.....' failed"

Do you guys have any idea what am I doing wrong?

**Comment by ajnart** [comment]
The default port for websockets seems to be 6001 according to the docker-compose file that has the `coolify-realtime` container. I haven't gotten it to work on my side either tho.

**Comment by ajnart** [comment]
After more tests @gmrizzo I got it to work by setting `PUSHER_PORT=` (to nothing) then it connects `wss://realtime.mydomain.com` instead of `wss://realtime.mydomain.com:PORT` and that works. I've also had to open the SSH port locally

**Reply by gmrizzo** [reply]
I changed `PUSHER_PORT` variable to nothing but it shows an error. What ssh port did you opened locally? On you OS firewall or did you add an ssh port mapping to docker-compose? 

Cloudflare tunnel conf:
*.mydomain.com => http://localhost:80 (as stated in the documentation)

As I understand it, in wildcard setup the request `realtime.example.com` goes to localhost:80 through the tunnel and should be handled by coolify-proxy, which should redirect the request to the coolify-realtime docker container. It seems that this last step doesn't work for me. 

I will take a closer look...

**Reply by gmrizzo** [reply, marked as answer]
Got it working! So in the end, my proxy config would not redirect the realtime connections to the right container. I previously added a config to cloudflare for realtime.mydomain.com => http://localhost:6001 to circumvent the proxy but this didn't work.
The key reason why this didn't work was the order of rules in cloudflare. My current config looks like this. Check if the order of hostnames is right.

- ssh.mydomain.com => ssh://localhost:22
- realtime.mydomain.com => http://localhost:6001
- *.mydomain.com => http://localhost:80
- mydomain.com => http://localhost:8000

I hope this helps anybody to fix their issues in the future with this topic.
